plugins/secretstores/docker/docker.go

```diff
@@ -0,0 +1,83 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package docker
+
+import (
+	_ "embed"
+	"errors"
+	"os"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/secretstores"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+// mount directory created by Docker when using Docker Secrets
+const dockerSecretsDir string = "/run/secrets"
+
+type Docker struct {
+	ID string `toml:"id"`
+}
+
+func (*Docker) SampleConfig() string {
+	return sampleConfig
+}
+
+// Init initializes all internals of the secret-store
+func (d *Docker) Init() error {
+	if d.ID == "" {
+		return errors.New("id missing")
+	}
+	if _, err := os.Stat(dockerSecretsDir); os.IsNotExist(err) {
+		return errors.New("/run/secrets directory does not exist")
+	}
+	return nil
+}
+
+func (d *Docker) Get(key string) ([]byte, error) {
+	secretFile := dockerSecretsDir + "/" + key
+	value, err := os.ReadFile(secretFile)
+	if err != nil {
+		return nil, errors.New("cannot find the secret file under /run/secrets")
+	}
+	if string(value) == "" {
+		return nil, errors.New("the value of the secrets file is empty")
+	}
+	return value, nil
+}
+
+func (d *Docker) List() ([]string, error) {
+	secrets := make([]string, 0)
+	secretFiles, err := os.ReadDir(dockerSecretsDir)
+	if err != nil {
+		return nil, errors.New("cannot read files under /run/secrets directory")
+	}
+	if len(secretFiles) == 0 {
+		return nil, errors.New("cannot find any secrets under /run/secrets")
+	}
+	for _, entry := range secretFiles {
+		secrets = append(secrets, entry.Name())
+	}
```

Comment:
This will be more efficient
```suggestion
	secretFiles, err := os.ReadDir(dockerSecretsDir)
	if err != nil {
		return nil, errors.New("cannot read files under /run/secrets directory")
	}
	secrets := make([]string, 0, len(secretFiles))
	for _, entry := range secretFiles {
		secrets = append(secrets, entry.Name())
	}
```
Furthermore, I don't think it is an error to have no secrets.

Reply:
I would like to reconsider the case when there are no secrets, if you are using this secret-store with docker and have forgotten to mention the secrets configuration in your compose file and you try to use it in the configuration resolution, then there will be errors. If it is not addressed here, there will be connectivity problems with plugins and will be difficult to troubleshoot.

In a nutshell, if you  forget to configure, it might help to let the user know no such secret ever existed in the first place and it might not be a `telegraf` problem

Reply:
@Hipska this is also a consideration when trying to check `/run/secrets` directory

Reply:
I don't think so, as here you specifically ask to list the available secrets. If the dir cannot be read because of it doesn't exist, then that counts as an error in this scenario to me..

Reply:
@Hipska I agree, if there are files but they cannot be read that's clearly an error!

@shantanoo-desai if there are no secrets and the secrets are not referenced than I do not see a problem (i.e. there should be no error). If a secret is referenced but not there, Telegraf will report this. If there are issues reading the secrets, see my comment above.